### ISSUE TITLE
NodeJS 6 backwards compatible support

### ISSUE THREAD

**Issue by jmendiara** [telefonicaid/logops#36]
Tests are failing for node6, mostly for some changes in how dates and errors are now default printed by `util.format`

**Comment by jmendiara** [comment]
It's... complicated. Node6 incorporated these breaking changes that we inherit (ref https://github.com/nodejs/node/wiki/Breaking-changes-between-v5-and-v6#util)
 * [util: Change how Error objects are formatted](https://github.com/nodejs/node/pull/4582)
 * [util,test: Use consistent Date representation for util/inspect](https://github.com/nodejs/node/pull/4318)

As we delegate in utils.format per design and to be developer friendly 
> With the rest of arguments is just like calling console.log. It will be serialized as the trace message. Easy to remember.

My proposal is

1. Make a 2.X release to npm making node<6 compatible. `package.json`
```json
{ "engines" : { "node" : "<6" } }
```
2. Deprecate logops v2.X npm releases in favour to the new 2.x release to warn the user about the issue (https://docs.npmjs.com/cli/deprecate)
3. Create logops v3 with node6 compatible engine
```json
{ "engines" : { "node" : ">=6" } }
```

@gtorodelvalle do you still want the `pipe` format to be maintained for this new v3 release?
CC/ @palmerabollo and @gtorodelvalle

**Comment by palmerabollo** [comment]
Your proposal sounds very good to me.

Another option, if we don't want to maintain the `pipe` format in v2, would be to give up and use https://github.com/trentm/node-bunyan (even if it's slower than logops). I guess our utilities in the Telefonica ecosystem (Telefonica/node-express-logging, etc) would keep working, since they only use simple logger.xxx functions. I don't know, I have a personal attachment to logops :)

**Comment by jmendiara** [comment]
I personally like some logops features that bunyan does not have:
* ENV_VARS configuration
* Clean integration with errors logging that have `cause`(eg: [Therror](https://github.com/therror/therror))
* Performance
* Usage without instantiation
* DEV format without piping to another utility
* Not printing these opinionated `hostname` and `pid` that does not have sense in modern/dockerized environments

Also, having `2,175 downloads in the last month` although is not too much, keeps me thinking that these [15 **public** packages](https://www.npmjs.com/browse/depended/logops ) deserve some extra effort from my side

I'll keep maintaining this

**Comment by jmendiara** [comment]
Closed by #39
